db-mongo/hiCapSeeder.js

```diff
@@ -0,0 +1,41 @@
+const faker = require('faker');
+const { createReadStream, createWriteStream } = require('graceful-fs')
+
+function generateFakeReviews (itemQuantity, filename) {
+  let header = createWriteStream(filename);
+  header.write('id,product_id,likes,stars,title,author,create_date,body,recommended\n');
+  let counter = 0;
+
+  const reviewsWriteStream = createWriteStream(filename, {flags: 'a'});
+
+
+  (async () => {
+      for (i = 0; i < itemQuantity; i++) {
+        let randomNumberOfReviews = Math.floor(Math.random() * 5) + 1;
+
+        for (j = randomNumberOfReviews; j > 0; j--) {
+          let id = counter + 1;
+          let productId = i + 1;
+          let fakeLikes = faker.random.number(105);
+          let fakeStars = faker.random.number(5);
+          let fakeTitle = faker.lorem.sentence();
+          let fakeAuthor = faker.name.findName();
+          let fakeDate = faker.date.recent();
+          let fakeBody = faker.lorem.paragraph();
+          let fakeRecommended = faker.random.boolean();
+
+          let review = `${id},${productId},${fakeLikes},${fakeStars},"${fakeTitle}","${fakeAuthor}","${fakeDate}","${fakeBody}",${fakeRecommended}\n`;
+          counter ++;
+          if (!reviewsWriteStream.write(review)) {
+            await new Promise(resolve => reviewsWriteStream.once('drain', resolve));
```

Comment:
This is a really interesting way to pause the for loop until the write stream drains.  I like it!

Reply:
+1 To this!

Reply:
I wish I could take credit for it! It was part of an article that helped me figure out how to get drain working.  :D